tests/reporter/kubernetes.go

```diff
@@ -918,3 +1024,157 @@ func writeStringToFile(filePath string, data string) error {
 	_, err = f.WriteString(data)
 	return err
 }
+
+func getVmiType(vmi v12.VirtualMachineInstance) (string, error) {
+	for _, volume := range vmi.Spec.Volumes {
+		if volume.VolumeSource.ContainerDisk == nil {
+			continue
+		}
+
+		image := volume.VolumeSource.ContainerDisk.Image
+		if strings.Contains(image, "fedora") {
+			return "fedora", nil
+		} else if strings.Contains(image, "cirros") {
+			return "cirros", nil
+		} else if strings.Contains(image, "alpine") {
+			return "alpine", nil
+		}
+	}
+
+	return "", fmt.Errorf("unknown type, vmi %s", vmi.ObjectMeta.Name)
+}
+
+func prepareVmiConsole(vmi v12.VirtualMachineInstance, vmiType string) error {
+	err := console.SafeExpectBatch(&vmi, []expect.Batcher{
+		&expect.BSnd{S: "\n"},
+		&expect.BExp{R: console.PromptExpression},
+	}, 5)
+
+	if err != nil {
+		switch vmiType {
+		case "fedora":
+			return console.LoginToFedora(&vmi)
+		case "cirros":
+			return console.LoginToCirros(&vmi)
+		case "alpine":
+			return console.LoginToAlpine(&vmi)
+		}
+	}
+
+	return err
+}
+
+func podHasComputeContainer(pod v1.Pod) bool {
+	for _, container := range pod.Spec.Containers {
+		if container.Name == "compute" {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (r *KubernetesReporter) executeNodeCommands(virtCli kubecli.KubevirtClient, logsdir string, pod v1.Pod) {
+	const networkPrefix = "nsenter -t 1 -n -- "
+	hostPrefix := fmt.Sprintf("%s --mount %s exec -- ", virt_chroot.GetChrootBinaryPath(), virt_chroot.GetChrootMountNamespace())
+
+	cmds := []commands{
+		{command: networkPrefix + "ip address", fileNameSuffix: "ipaddress"},
+		{command: networkPrefix + "ip link", fileNameSuffix: "iplink"},
+		{command: networkPrefix + "ip route show table all", fileNameSuffix: "iproute"},
+		{command: networkPrefix + "ip neigh show", fileNameSuffix: "ipneigh"},
+		{command: networkPrefix + "bridge -j vlan show", fileNameSuffix: "brvlan"},
+		{command: networkPrefix + "bridge fdb", fileNameSuffix: "brfdb"},
+		{command: networkPrefix + "nft list ruleset", fileNameSuffix: "nftlist"},
+
+		{command: hostPrefix + "/usr/bin/" + networkPrefix + "/usr/sbin/iptables --list", fileNameSuffix: "iptables"},
```

Comment:
I think the interesting part in the iptables output is the counters. I would add the `-v` option.